src/Raven.Server/Documents/ChangesClientConnection.cs

```diff
@@ -383,6 +391,12 @@ public void SendDocumentChanges(DocumentChange change)
 
         public void SendIndexChanges(IndexChange change)
         {
+            if (_aggressiveChanges)
+            {
+                PulseAggressiveCaching();
```

Comment:
can we have here Debug.Assert at least that `_watchAllIndexes == 0 && (_matchingIndexes == null || _matchingIndexes.Count == 0)` ?

Reply:
Done